submissions/misshatsker/object-oriented-js/app.js

```diff
@@ -0,0 +1,188 @@
+const leftSide = 0;
+const topSide = 0;
+const rightSide = 500;
+const bottomSide = 500;
+
+const initialGroundY = 60;
+const tileHeight = 80;
+
+const enemySprite = 'images/enemy-bug.png';
+const enemyWidth = 80;
+const enemyHeight = 70;
+const enemyMinSpeed = 100;
+const enemyMaxSpeed = 300;
+const enemyMinX = -rightSide;
+const enemyMaxX = enemyWidth - rightSide;
+
+const playerSprite = 'images/char-princess-girl.png';
+const playerWidth = 70;
+const playerHeight = 70;
+const playerSpeed = 75;
+const playerInitialX = 200;
+const playerInitialY = 400;
+
+const Character = function(x, y, speed, sprite) {
+    this.sprite = sprite;
+    this.speed = speed;
+    this.x = x;
+    this.y = y;
+};
+
+Character.prototype.render = function() {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+const Enemy = function(...args) {
```

Comment:
Bad practice.
IDE intellisense won't be able to hint what arguments are expected.
Is there any good reason to have `...args`?
Why not using named parameters?

Reply:
Just to not copy/paste them as far as I wanted to reuse the `Character` constructor completely. But I got your notice with IDE, thanks! I haven't thought about that